testing/InstallTestbed.ps1

```diff
@@ -0,0 +1,338 @@
+Write-Output $ProcessSeparator
+
+# Source the variables from the file
+Write-Output "`nSourcing the variables from the file..."
+. ./configure/azure_scripts/config.ps1
+
+# Remove old code if it exists
+if (Test-Path ./configure.zip) {
+    Remove-Item ./configure.zip -Force -Confirm:$false -ErrorAction SilentlyContinue
+} else {
+    Write-Output "File not found."
+}
+Write-Output $ProcessSeparator
+
+# Zip up the installer scripts for the VM
+Write-Output "`nZipping up the installer scripts for the VM..."
+./configure/azure_scripts/zip_my_parents_parent.ps1
+Write-Output $ProcessSeparator
+
+# Upload the zip file to the container and get a key to download it
+Write-Output "`nUploading the zip file to the container and getting a key to download it..."
+$FileDownloadUrl = ./configure/azure_scripts/copy_file_to_container.ps1 `
+    -LocalFilePath "configure.zip" `
+    -ContainerName $ContainerName `
+    -StorageAccountName $StorageAccountName `
+    -StorageAccountKey $StorageAccountKey
+
+Write-Output "File download URL: $FileDownloadUrl"
+Write-Output $ProcessSeparator
+
+Write-Output "`nChanging directory to the azure scripts..."
+Set-Location configure/azure_scripts
+Write-Output $ProcessSeparator
+
+# Make our directory on the VM
+Write-Output "`nMaking our directory on the VM..."
+$createDirResponse = az vm run-command invoke `
+  --command-id RunPowerShellScript `
+  --name $VMName `
+  --resource-group $ResourceGroup `
+  --scripts "if (-not (Test-Path -Path 'C:\lme')) { New-Item -Path 'C:\lme' -ItemType Directory }"
+Show-FormattedOutput -FormattedOutput (Format-AzVmRunCommandOutput -JsonResponse "$createDirResponse")
+Write-Output $ProcessSeparator
+
+
+# Download the zip file to the VM
+Write-Output "`nDownloading the zip file to the VM..."
+$downloadZipFileResponse = .\download_in_container.ps1 `
+    -VMName $VmName `
+    -ResourceGroup $ResourceGroup `
+    -FileDownloadUrl "$FileDownloadUrl" `
+    -DestinationFilePath "configure.zip"
+Show-FormattedOutput -FormattedOutput (Format-AzVmRunCommandOutput -JsonResponse "$downloadZipFileResponse")
+Write-Output $ProcessSeparator
+
+# Extract the zip file
+Write-Output "`nExtracting the zip file..."
+$extractArchiveResponse = .\extract_archive.ps1 `
+    -VMName $VMName `
+    -ResourceGroup $ResourceGroup `
+    -FileName "configure.zip"
+Show-FormattedOutput -FormattedOutput (Format-AzVmRunCommandOutput -JsonResponse "$extractArchiveResponse")
+Write-Output $ProcessSeparator
+
+# Run the install script for chapter 1
+Write-Output "`nRunning the install script for chapter 1..."
+$installChapter1Response = .\run_script_in_container.ps1 `
+    -ResourceGroup $ResourceGroup `
+    -VMName $VMName `
+    -ScriptPathOnVM "C:\lme\configure\install_chapter_1.ps1"
+Show-FormattedOutput -FormattedOutput (Format-AzVmRunCommandOutput -JsonResponse "$installChapter1Response")
+Write-Output $ProcessSeparator
+
+# Todo: Loop these for number of vms
```

Comment:
Is this Todo still needed? If I'm reading `Invoke-GPUpdateOnVMs` correctly it is looping through the clients already.
```suggestion
```

Reply:
Done